src/main/java/bio/terra/app/usermetrics/BardClient.java

```diff
@@ -56,4 +78,69 @@ public void logEvent(AuthenticatedUserRequest userReq, BardEvent event) {
       logger.warn("Error logging event {}", event.getEvent(), e);
     }
   }
+
+  /**
+   * Syncing a user is only needed when a new token is not already cached or the cache entry has
+   * expired.
+   *
+   * @param userReq - the AuthenticatedUserRequest that represents the current user.
+   */
+  boolean syncUser(AuthenticatedUserRequest userReq) {
+    boolean updatedEntry = false;
+    String key = userReq.getToken();
+    if (!bearerCache.containsKey(key)) {
```

Comment:
I think this might be a race condition if you have multiple calls.  You could use `bearerCache.computeIfAbsent`.  Is the reason you didn't since you don't want to add the entry if `syncProfile` fails?  Alternately, you could put the check/set in a synchronized block.

Reply:
It does need to be synchronized because I can't use the computeIfAbsent.  I'll make that change.